### DISCUSSION TITLE
close_action trap with a NAT setup

### DISCUSSION THREAD

**Discussion by misch42** [strongswan/strongswan#2183] [General]
Hi,

we have a VPN setup with a POSTROUTING NAT rule to translate our source address to a transform selector that the other side accepts. The other side from time to time closes the IKE_SA so I configured the close_action=trap. So my side should initiate the tunnel again.

But that somehow does not happen. The tunnel stays down.

The important sections of the config are:
me-you-child {
            local_ts = 10.110.0.0/24
            remote_ts = 192.168.1.0/24
            start_action = start
            close_action = trap
         }

and

-A POSTROUTING -s 172.16.0.0/24 -d 192.168.1.0/24 -j SNAT --to-source 10.110.0.253

When I do a ping from my side and do a tcpdump I see the packets comming in and beeing NATted_

IP 172.16.0.1 > 192.168.1.5: ICMP echo request, id 37970, seq 1867, length 64
IP 10.110.0.253 > 192.168.1.5: ICMP echo request, id 37970, seq 1867, length 64

but the trap does not initiate the tunnel.

Is there a problem with the XFRM policies and the POSTROUTING chain?

Michael.

**Comment by tobiasbrunner** [comment]
> Is there a problem with the XFRM policies and the POSTROUTING chain?

Since you haven't posted what policies are installed at that time, no idea. Are there even trap policies installed? Also check the log for what happens when the peer terminates the connection.

Note that `close_action` only has an effect when the SA is actively closed, not when it gets closed due to failed retransmits or some other reason e.g. a failed rekeying. If you use a new enough version, you can also configure `start_action = start|trap` so you'll have the trap policy installed from the beginning.
